src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp

```diff
@@ -615,18 +723,28 @@ Strings IcebergMetadata::getDataFiles() const
             if (ManifestEntryStatus(status) == ManifestEntryStatus::DELETED)
             {
                 LOG_TEST(log, "Processing delete file for path: {}", file_path);
-                chassert(!files.contains(file_path));
+                chassert(files.contains(file_path) == 0);
             }
             else
             {
                 LOG_TEST(log, "Processing data file for path: {}", file_path);
-                files.insert(file_path);
+                files[file_path] = schema_object_id;
             }
         }
+
+        schema_processor.addIcebergTableSchema(schema_object);
```

Comment:
Why do we need it? AFAIU `schema_processor` should already contain all the schemas after `parseTableSchema`

Reply:
Ah, I believe it's because of version 1 where we might not have `schemas` field in the metadata.